lib/levelup.js

```diff
@@ -47,20 +47,30 @@ function LevelUP (location, options, callback) {
     return new LevelUP(location, options, callback)
 
   var error
+    , down
 
   EventEmitter.call(this)
   this.setMaxListeners(Infinity)
 
-  if (typeof location == 'function') {
+  if (isLevelDOWN(location)) {
+    options = typeof options == 'object' ? options : {}
+    down = location
+    options.db = function () { return down }
```

Comment:
I'm all for simplifying, but I'd prefer to do all of that in a separate PR. I guess this change will involve a new major?